homeassistant/components/whirlpool/climate.py

```diff
@@ -0,0 +1,232 @@
+"""Platform for climate integration."""
+import logging
+
+from whirlpool.aircon import Aircon, FanSpeed as AirconFanSpeed, Mode as AirconMode
+from whirlpool.auth import Auth
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_OFF,
+    HVAC_MODE_COOL,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_SWING_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+    SWING_HORIZONTAL,
+    SWING_OFF,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from .const import AUTH_INSTANCE_KEY, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+AIRCON_MODE_MAP = {
+    AirconMode.Cool: HVAC_MODE_COOL,
+    AirconMode.Heat: HVAC_MODE_HEAT,
+    AirconMode.Fan: HVAC_MODE_FAN_ONLY,
+}
+
+AIRCON_FANSPEED_MAP = {
+    AirconFanSpeed.Off: FAN_OFF,
+    AirconFanSpeed.Auto: FAN_AUTO,
+    AirconFanSpeed.Low: FAN_LOW,
+    AirconFanSpeed.Medium: FAN_MEDIUM,
+    AirconFanSpeed.High: FAN_HIGH,
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up entry."""
+    auth: Auth = hass.data[DOMAIN][config_entry.entry_id][AUTH_INSTANCE_KEY]
+    said_list = auth.get_said_list()
+    if not said_list:
+        _LOGGER.debug("No appliances found")
+        return
+
+    aircon = AirConEntity(said_list[0], auth)
+    async_add_entities([aircon], True)
+
+
+class AirConEntity(ClimateEntity):
+    """Representation of an air conditioner."""
+
+    def __init__(self, said, auth: Auth):
+        """Initialize the entity."""
+        self._aircon = Aircon(auth, said, self.schedule_update_ha_state)
+
+        self._name = said
+        self._supported_features = (
+            SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE | SUPPORT_SWING_MODE
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Connect aircon to the cloud."""
+        await self._aircon.connect()
+
+        try:
+            name = await self._aircon.fetch_name()
+            if name is not None:
+                self._name = name
+        except Exception:  # pylint: disable=broad-except
```

Comment:
Only catch the exceptions that `self._aircon.fetch_name` is expected to throw.

Reply:
done